### PR TITLE
Use bounded channels for real-time safety

### LINKED ISSUE #304: Do not use unbounded channels in online audio processing
As mentioned in https://github.com/orottier/web-audio-api-rs/issues/297:

> I also consider using unbounded channels for this purpose as inappropriate. For any for real world use case bounded channels are recommended. Moreover the receiver side of theses crossbeam channels cannot be used safely in a real-time context AFAIK.

> The crossbeam channels seem to have serious unresolved issues: https://github.com/crossbeam-rs/crossbeam/issues/821

The `AudioContext` (as opposed to the `OfflineAudioContext`) should only use bounded channels.
It is probably okay for the control thread to block on a full channel.
It is probably okay for the render thread to drop data when a channel is full (we use them for events and metrics).

### PR DESCRIPTION
**Pull request by orottier**
Fixes #304

### CODE REVIEW

File: src/io/mod.rs [hunk 1]
```diff
@@ -42,14 +42,23 @@ pub(crate) struct RenderThreadInit {
...
 
 pub(crate) fn thread_init() -> (ControlThreadInit, RenderThreadInit) {
-    // track number of frames - synced from render thread to control thread
+    // Track number of frames - synced from render thread to control thread
     let frames_played = Arc::new(AtomicU64::new(0));
-    // communication channel for ctrl msgs to the render thread
-    let (ctrl_msg_send, ctrl_msg_recv) = crossbeam_channel::unbounded();
-    // communication channel for render load values
+
+    // Communication channel for ctrl msgs to the render thread.
```
Comment (b-ma) [inline src/io/mod.rs:48]:
```
// Communication channel for ctrl msgs from the control thread to the render thread.
```

File: src/io/mod.rs [hunk 2]
```diff
@@ -42,14 +42,23 @@ pub(crate) struct RenderThreadInit {
...
+    // the capacity is reached.
+    let (ctrl_msg_send, ctrl_msg_recv) = crossbeam_channel::bounded(256);
+
+    // Communication channel for render load values.
+    // A dedicated thread is consuming these messages so there is no need for buffering.
     let (load_value_send, load_value_recv) = crossbeam_channel::bounded(1);
-    // communication channel for events for render thread to control thread
-    let (event_send, event_recv) = crossbeam_channel::unbounded();
+
+    // Communication channel for events for render thread to control thread.
```
Comment (b-ma) [inline src/io/mod.rs:58]:
```
// Communication channel for events from the render thread to the control thread.
```

**Comment by b-ma** [comment]
Apart my small comments on the comments, looks all good to me (and nice catch with callback / mutex issue)!

File: src/io/mod.rs (hunk 1 above)
Comment (orottier) [inline src/io/mod.rs:48]:
```suggestion
    // Communication channel for ctrl msgs from the control thread to the render thread.
```

File: src/io/mod.rs (hunk 2 above)
Comment (orottier) [inline src/io/mod.rs:58]:
```suggestion
    // Communication channel for events from the render thread to the control thread.
```

**Comment by orottier** [comment]
Thanks for the feedback, I applied the changes.
Did you know that for these small changes, you can suggest the edit in code directly: https://egghead.io/lessons/github-add-suggestions-in-a-github-pr-review
Saves me a few small steps :)

I will merge this and put out a new release

**Comment by b-ma** [comment]
> Did you know that for these small changes, you can suggest the edit in code directly: https://egghead.io/lessons/github-add-suggestions-in-a-github-pr-review
> Saves me a few small steps :)

Oh ok didn't know that

> I will merge this and put out a new release

Cool thanks!

**Pull request merged**